src/include/common/printable.h

```diff
@@ -1,25 +1,54 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Terrier
+//
+// printable.h
+//
+// Identification: src/include/common/printable.h
+//
+// Copyright (c) 2015-18, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
 #pragma once
 
 #include <iosfwd>
 #include <string>
 
+#ifdef __APPLE__
+#include <json/json.h>
+#else
+#include <jsoncpp/json/json.h>
+#endif
+
 namespace terrier {
+
+//===--------------------------------------------------------------------===//
+// Printable Object
+//===--------------------------------------------------------------------===//
+
 /**
- * Stateless interface to allow printing of debug information
+ * @brief Interface to allow printing of debug information
  *
  * Most stateful classes should implement this interface and return useful
  * information about its state for debugging purposes.
  */
 class Printable {
  public:
-  virtual ~Printable() = default;
+  virtual ~Printable(){};
+
+  /** @brief Set the Json value about the printable object. */
+  virtual void SetPrintable() = 0;
+
+  /** @brief Get the Json value about the printable object. */
+  Json::Value GetPrintable ();
 
-  /** @brief Return information about the object in human-readable way. */
-  virtual const std::string GetInfo() const = 0;
+  /** @brief Get the styled string from the Json value. */
+  std::string GetInfo ();
 
-  friend std::ostream &operator<<(std::ostream &os, const Printable &printable) {
```

Comment:
Why did you get rid of this?

Reply:
We don't need it anymore since we can get the embedding feature and styled string for free once we have json values.

Reply:
JSON library also overload the `<<` operator even if we want to directly call it, which is not necessary though.